libtiledbsoma/src/soma/soma_array.cc

```diff
@@ -1141,6 +1140,105 @@ std::optional<TimestampRange> SOMAArray::timestamp() {
+        names, tiledb_datatypes);
+
+    // Create the data for the info we return
+    auto arrow_array = ArrowAdapter::make_arrow_array_parent(array_ndim);
+
+    for (int i = 0; i < array_ndim; i++) {
+        auto core_dim = dimensions[i];
+        auto core_type_code = core_dim.type();
+
+        ArrowArray* child = nullptr;
+
+        switch (core_type_code) {
+            case TILEDB_INT64:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<int64_t>(core_dim.name(), which_kind));
+                break;
+            case TILEDB_UINT64:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<uint64_t>(
+                        core_dim.name(), which_kind));
+                break;
+            case TILEDB_INT32:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<int32_t>(core_dim.name(), which_kind));
+                break;
+            case TILEDB_UINT32:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<uint32_t>(
+                        core_dim.name(), which_kind));
+                break;
+            case TILEDB_INT16:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<int16_t>(core_dim.name(), which_kind));
+                break;
+            case TILEDB_UINT16:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<uint16_t>(
+                        core_dim.name(), which_kind));
+                break;
+            case TILEDB_INT8:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<int8_t>(core_dim.name(), which_kind));
+                break;
+            case TILEDB_UINT8:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<uint8_t>(core_dim.name(), which_kind));
+                break;
+
+            case TILEDB_FLOAT64:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<double>(core_dim.name(), which_kind));
+                break;
+            case TILEDB_FLOAT32:
+                child = ArrowAdapter::make_arrow_array_child(
+                    _core_domainish_slot<float>(core_dim.name(), which_kind));
+                break;
+
+            case TILEDB_STRING_ASCII:
+            case TILEDB_CHAR:
+                child = ArrowAdapter::make_arrow_array_child_string(
+                    _core_domainish_slot_string(core_dim.name(), which_kind));
+                break;
+
+            default:
+                break;
+        }
+
+        if (child == nullptr) {
+            // throw TileDBSOMAError(fmt::format(
+            //   "WIP {} {}",
+            //   core_dim.name(),
+            //   tiledb::impl::type_to_str(core_type_code)));
+        }
```

Comment:
Did you intend to throw here if the `core_type_code` was not found or set the child to `nullptr`?

Reply:
derp sorry, that's a WIP I didn't finish & I shouldn't have put this PR up for review yet

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/3012/commits/804db2b41d3c4f31e41fd08bbc55b5730c79a794